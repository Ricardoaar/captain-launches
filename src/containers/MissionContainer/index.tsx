import React from "react";

type Props = {
  children: JSX.Element | JSX.Element[];
};
const MissionContainer: React.FC<Props> = ({ children }) => {
  return <div className="mission-container ">{children}</div>;
};

export default MissionContainer;
