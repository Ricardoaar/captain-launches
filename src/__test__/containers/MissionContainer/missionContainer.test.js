import { create } from "react-test-renderer";

const { mount } = require("enzyme");
import MissionContainer from "../../../containers/MissionContainer";

describe("MissionContainer", () => {
  const wrapper = mount(
    <MissionContainer>
      <h1>Cats</h1>
    </MissionContainer>
  );
  test("should render correctly", () => {
    expect(wrapper.length).toBe(1);
  });
  test("should render a div", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });
  test("should render children", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });
  test("should match snapshot", () => {
    const snapWrapper = create(
      <MissionContainer>
        <h1>Cats</h1>
      </MissionContainer>
    );
    expect(snapWrapper.toJSON()).toMatchSnapshot();
  });
});
