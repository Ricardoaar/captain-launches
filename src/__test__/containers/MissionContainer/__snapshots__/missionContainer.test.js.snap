// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionContainer should match snapshot 1`] = `
<div
  className="mission-container "
>
  <h1>
    Cats
  </h1>
</div>
`;
